### PR TITLE
python: Properly report HTTPExceptions

### PR DESCRIPTION
**Pull request by martinpitt**
If reading the response throws a http.client exception, close the channel with that error message, instead of crashing the thread and letting it log tracebacks. This does not change observable behaviour much, but cleans up unexpected error messages.

---

This will fix the last [podman failure](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1196-20230202-124820-1dc12492-fedora-37-pybridge/log.html) with the pybridge, see https://github.com/cockpit-project/cockpit-podman/pull/1196

### CODE REVIEW

File: src/cockpit/channels/http.py [hunk 1]
```diff
@@ -108,11 +108,19 @@ def request(self):
...
-            headers=self.parse_headers(response.headers)))
-        self.read_send_response(response)
+        try:
+            response = connection.getresponse()
+            self.loop.call_soon_threadsafe(lambda: self.send_control(
+                command='response', status=response.status, reason=response.reason,
+                headers=self.parse_headers(response.headers)))
+            self.read_send_response(response)
+        except http.client.HTTPException as error:
+            logger.debug('HTTP %s reading response failed: %s', error)
```
Comment (jelly) [inline src/cockpit/channels/http.py:118]: This passes one argument while the format string expects two? Also shouldn't we call `str(error`)

**Review by jelly — DISMISSED** [review]

File: src/cockpit/channels/http.py [hunk 2]
```diff
@@ -108,11 +108,19 @@ def request(self):
...
+            response = connection.getresponse()
+            self.loop.call_soon_threadsafe(lambda: self.send_control(
+                command='response', status=response.status, reason=response.reason,
+                headers=self.parse_headers(response.headers)))
+            self.read_send_response(response)
+        except http.client.HTTPException as error:
+            msg = str(error)
+            logger.debug('HTTP reading response failed: %s', msg)
+            self.loop.call_soon_threadsafe(lambda: self.close(problem='terminated', message=msg))
+            connection.close()
```
Comment (allisonkarlitskaya) [inline src/cockpit/channels/http.py:121]: This goes in `finally:`

**Review by allisonkarlitskaya — CHANGES_REQUESTED** [review]

File: src/cockpit/channels/http.py (hunk 2 above)
Comment (martinpitt) [reply src/cockpit/channels/http.py]: Thanks! Done.

**Review by allisonkarlitskaya — APPROVED** [review]
:+1: thx

**Pull request merged**